From ed57c661ed3bf26e248d252cb9c110ba33440aeb Mon Sep 17 00:00:00 2001
From: Yuta Saito <kateinoigakukun@gmail.com>
Date: Wed, 7 Jun 2023 19:04:23 +0000
Subject: [PATCH] [static-stdlib] Move static-executable-args.lnk under
 stdlib/public/Resources/linux

This is a preparatory change for adding a static executable support for WASI
---
 .../public/Resources/linux}/static-executable-args.lnk       | 0
 stdlib/public/runtime/CMakeLists.txt                         | 5 +++--
 2 files changed, 3 insertions(+), 2 deletions(-)
 rename {utils => stdlib/public/Resources/linux}/static-executable-args.lnk (100%)

diff --git a/utils/static-executable-args.lnk b/stdlib/public/Resources/linux/static-executable-args.lnk
similarity index 100%
rename from utils/static-executable-args.lnk
rename to stdlib/public/Resources/linux/static-executable-args.lnk
diff --git a/stdlib/public/runtime/CMakeLists.txt b/stdlib/public/runtime/CMakeLists.txt
index 0d3bc17f595..e43c0775cdb 100644
--- a/stdlib/public/runtime/CMakeLists.txt
+++ b/stdlib/public/runtime/CMakeLists.txt
@@ -114,18 +114,19 @@ set(sdk "${SWIFT_HOST_VARIANT_SDK}")
 if(SWIFT_BUILD_STATIC_STDLIB AND "${sdk}" STREQUAL "LINUX")
   set(static_binary_lnk_file_list)
   string(TOLOWER "${sdk}" lowercase_sdk)
+  set(static_binary_lnk_src "${SWIFT_SOURCE_DIR}/stdlib/public/Resources/${lowercase_sdk}/static-executable-args.lnk")
 
   # Generate the static-executable-args.lnk file used for ELF systems (eg linux)
   set(linkfile "${lowercase_sdk}/static-executable-args.lnk")
   add_custom_command_target(swift_static_binary_${sdk}_args
     COMMAND
       "${CMAKE_COMMAND}" -E copy
-      "${SWIFT_SOURCE_DIR}/utils/static-executable-args.lnk"
+      "${static_binary_lnk_src}"
       "${SWIFTSTATICLIB_DIR}/${linkfile}"
     OUTPUT
       "${SWIFTSTATICLIB_DIR}/${linkfile}"
     DEPENDS
-      "${SWIFT_SOURCE_DIR}/utils/static-executable-args.lnk")
+      "${static_binary_lnk_src}")
 
   list(APPEND static_binary_lnk_file_list ${swift_static_binary_${sdk}_args})
   swift_install_in_component(FILES "${SWIFTSTATICLIB_DIR}/${linkfile}"
-- 
2.40.0

